### PR TITLE
Specifiy output file name

### LINKED ISSUE #220: Ability to specify filename prefix and/or suffix
## Description

Is it possible to add an input parameter to allow the user to add some relevant information to the output filename?
For example, when running ensemble_grid_point_* with Average=True leads to files lacking the required info for the user to associate which result goes with which request. For example it will make more sense if the output files names starts with:  "health_region-de-montréal" instead of "ensemble_grid_point".

### PR DESCRIPTION
**Pull request by aulemahal**
## Overview

This PR fixes #220

Changes:

* Added argument `output_name` to ensemble process (it controls the prefix of the finale names) and to xclim and misc processes (where it controls to filename entirely). Did not add it to the subset/average processes where the filename is the same as the input.

### CODE REVIEW

**Comment by aulemahal** [comment]
D'intérêt pour @ldperron!

**Review by huard — APPROVED** [review]

File: finch/processes/utils.py [hunk 1]
```diff
@@ -713,3 +713,17 @@ def update_history(
         f"[{datetime.now():%Y-%m-%d %H:%M:%S}] {new_name or ''}: {hist_str} - finch version: {__version__}."
     )
     return merged_history
+
+
+def valid_filename(name):
```
[the remark this answers is not in the document]
Comment (aulemahal) [reply finch/processes/utils.py:718]: Good idea! This package seems quite comprehensive without much dependencies, it's a more robust solution.

[the remark this answers is not in the document]
Comment (ldperron) [reply finch/processes/utils.py:718]: Agreed, even better! Everything seems to be covered (  $(...)   `  .. . / )

**Comment by tlvu** [comment]
FYI Jenkins failure last night with this PR.  I guess a new Finch 0.8.2 release (https://github.com/bird-house/finch/pull/226) on PAVICS will fix this.

```
19:50:57  _________ finch-master/docs/source/notebooks/finch-usage.ipynb::Cell 1 _________
19:50:57  Notebook cell execution failed
19:50:57  Cell 1: Cell outputs differ
19:50:57  
19:50:57  Input:
19:50:57  help(wps.frost_days)
19:50:57  
19:50:57  Traceback:
19:50:57   mismatch 'stdout'
19:50:57  
19:50:57   assert reference_output == test_output failed:
19:50:57  
19:50:57    'Help on meth...ut files.\n\n' == 'Help on meth...ut files.\n\n'
19:50:57    Skipping 201 identical leading characters in diff, use -v to show
19:50:57    - iable=None) method of birdy.client.base.WPSClient instance
19:50:57    + iable=None, output_name=None) method of birdy.client.base.WPSClient instance
19:50:57    ?           ++++++++++++++++++
19:50:57          Number of days where daily minimum temperatures are below a threshold temperature.
19:50:57          
19:50:57          Parameters
19:50:57          ----------
19:50:57          tasmin : ComplexData:mimetype:`application/x-netcdf`, :mimetype:`application/x-ogc-dods`
19:50:57              NetCDF Files or archive (tar/zip) containing netCDF files. Minimum surface temperature.
19:50:57          thresh : string
19:50:57              Freezing temperature.
19:50:57          freq : {'YS', 'MS', 'QS-DEC', 'AS-JUL'}string
19:50:57              Resampling frequency.
19:50:57          check_missing : {'any', 'wmo', 'pct', 'at_least_n', 'skip', 'from_context'}string
19:50:57              Method used to determine which aggregations should be considered missing.
19:50:57          missing_options : ComplexData:mimetype:`application/json`
19:50:57              JSON representation of dictionary of missing method parameters.
19:50:57          cf_compliance : {'log', 'warn', 'raise'}string
19:50:57              Whether to log, warn or raise when inputs have non-CF-compliant attributes.
19:50:57          data_validation : {'log', 'warn', 'raise'}string
19:50:57              Whether to log, warn or raise when inputs fail data validation checks.
19:50:57          variable : string
19:50:57              Name of the variable in the NetCDF file.
19:50:57    +     output_name : string
19:50:57    +         Filename of the output (no extension).
19:50:57          
19:50:57          Returns
19:50:57          -------
19:50:57          output_netcdf : ComplexData:mimetype:`application/x-netcdf`
19:50:57              The indicator values computed on the original input grid.
19:50:57          output_log : ComplexData:mimetype:`text/plain`
19:50:57              Collected logs during process run.
19:50:57          ref : ComplexData:mimetype:`application/metalink+xml; version=4.0`
19:50:57              Metalink file storing all references to output files.
19:50:57      
19:50:57  
19:50:57  
19:50:57  _________ finch-master/docs/source/notebooks/finch-usage.ipynb::Cell 3 _________
19:50:57  Notebook cell execution failed
19:50:57  Cell 3: Cell outputs differ
19:50:57  
19:50:57  Input:
19:50:57  print("Process status: ", resp.status)
19:50:57  urls = resp.get()
19:50:57  print("Link to process output: ", urls.output_netcdf)
19:50:57  
19:50:57  
19:50:57  Traceback:
19:50:57   mismatch 'stdout'
19:50:57  
19:50:57   assert reference_output == test_output failed:
19:50:57  
19:50:57    'Process stat...20650101.nc\n' == 'Process stat...20650101.nc\n'
19:50:57    Skipping 94 identical leading characters in diff, use -v to show
19:50:57    - FILE/frost-days_SRES-A2-experiment_20460101-20650101.nc
19:50:57    ?           ^     ^^^^^^ ^                   ^
19:50:57    + FILE/frost_days_sres_a2_experiment_20460101_20650101.nc
19:50:57    ?           ^     ^^^^^^ ^                   ^
19:50:57  
```

**Pull request merged**